postgis/lwgeom_out_marc21.c

```diff
@@ -0,0 +1,406 @@
+/**********************************************************************
+ *
+ * PostGIS - Spatial Types for PostgreSQL
+ * http://postgis.net
+ *
+ * PostGIS is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * PostGIS is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with PostGIS.  If not, see <http://www.gnu.org/licenses/>.
+ **********************************************************************/
+
+#include "postgres.h"
+#include "fmgr.h"
+#include "utils/builtins.h"
+
+#include "../postgis_config.h"
+#include "liblwgeom.h"
+#include "lwgeom_transform.h"
+
+#include "stringbuffer.h"
+#include "liblwgeom_internal.h"
+
+#define MARC21_NS ((char *) "http://www.loc.gov/MARC21/slim")
+/**********************************************************************
+ * Ability to serialise MARC21/XML Records from geometries. Coordinates
+ * are returned in decimal degrees.
+ *
+ * MARC21/XML version supported: 1.1
+ * MARC21/XML Cartographic Mathematical Data Definition:
+ *    https://www.loc.gov/marc/bibliographic/bd034.html
+ *
+ * Copyright (C) 2021 University of Münster (WWU), Germany
+ * Written by Jim Jones <jim.jones@uni-muenster.de>
+ *
+ **********************************************************************/
+
+
+static int gbox_to_marc21_sb(const GBOX box, const char* format, stringbuffer_t *sb);
+static int is_format_valid(const char* format);
+lwvarlena_t* lwgeom_to_marc21(const LWGEOM *geom, const char* format);
+
+PG_FUNCTION_INFO_V1(ST_AsMARC21);
+Datum ST_AsMARC21(PG_FUNCTION_ARGS) {
+	lwvarlena_t *marc21;
+	int32_t srid;
+	LWPROJ *lwproj;
+	LWGEOM *lwgeom;
+	//uint8_t is_latlong;
+	GSERIALIZED *gs = PG_GETARG_GSERIALIZED_P_COPY(0);
```

Comment:
really need a copy? Are you muting it?

Reply:
Not really.. I just copied the wrong function ;) Thanks for the review!